File: 350. Intersection of Two Arrays II.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersect = function(nums1, nums2) {
    var res = [];
    var i1 = 0;
    var i2 = 0;
    nums1.sort(function(a, b){return a-b});
    nums2.sort(function(a, b){return a-b});
    while (i1 < nums1.length && i2 < nums2.length)
    {
        if (nums1[i1] < nums2[i2])
            i1 += 1;
        else if (nums1[i1] > nums2[i2])
            i2 += 1;
        else if (nums1[i1] == nums2[i2])
        {
            res.push(nums1[i1]);
            i1 += 1;
            i2 += 1;
        }
    }
    return res;
};

File: 350. Intersection of Two Arrays II.py
class Solution(object):
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        nums1, nums2 = sorted(nums1), sorted(nums2)
        i1, i2 = 0, 0
        res = []
        while i1 < len(nums1) and i2 < len(nums2):
            if nums1[i1] < nums2[i2]:
                i1 += 1
            elif nums1[i1] > nums2[i2]:
                i2 += 1
            elif nums1[i1] == nums2[i2]:
                res += [nums1[i1]]
                i1 += 1
                i2 += 1
        return res
